### PR TITLE
cli: fix for --target

### PR DESCRIPTION
**Pull request by roksys**
* New way to format url in case of $graph workflow.
  Solves #991.

Signed-off-by: Rokas Maciulaitis <rokas.maciulaitis@cern.ch>

### CODE REVIEW

**Comment by mr-c** [comment]
Jenkins, okay to test

**Comment by mr-c** [comment]
@roksys Thank you for this PR. This will needs at least one test. We can probably re-use `tests/wf/scatter-wf4.cwl`

File: cwltool/main.py [hunk 1]
```diff
@@ -13,6 +13,7 @@
 import sys
 import time
 from codecs import StreamWriter, getwriter  # pylint: disable=unused-import
+from urlparse import urlparse
```
Comment (mr-c) [inline cwltool/main.py]:
```suggestion
from six.moves import urllib
```

File: cwltool/main.py [hunk 2]
```diff
@@ -702,9 +703,13 @@ def my_represent_none(self, data):  # pylint: disable=unused-argument
 
             if args.target:
                 if isinstance(tool, Workflow):
-                    extracted = get_subgraph([document_loader.fetcher.urljoin(tool.tool["id"], "#"+r)
-                                              for r in args.target],
-                                             tool)
+                    url = urlparse(tool.tool["id"])
```
Comment (mr-c) [inline cwltool/main.py]:
```suggestion
                    url = urllib.parse.urlparse(tool.tool["id"])
```

**Comment by roksys** [comment]
Hello @mr-c,

Could you help me to understand why tests are failing?

https://ci.appveyor.com/project/mr-c/cwltool/builds/20678293/job/3rutn5qxpak8h9wo

**Comment by mr-c** [comment]
Hello @roksys. Appveyor does not yet support Linux Docker Containers, so many tests need decorating with `@windows_needs_docker` so that they are skipped there. Any execution of a CWL tool under MS Windows requires Docker.

**Pull request merged**